data-model/src/main/java/io/micronaut/data/model/query/builder/AbstractSqlLikeQueryBuilder.java

```diff
@@ -1031,7 +1031,8 @@ private void appendAdditionalWhere(StringBuilder queryClause, QueryState querySt
             while (matcher.find()) {
                 String name = matcher.group(3);
                 String placeholder = queryState.addAdditionalRequiredParameter(name);
-                matcher.appendReplacement(additionalWhereBuilder, placeholder);
+                //appendReplacement considers $ to be a special character, need to escape it.
```

Comment:
It would be better to have `Pattern#quote`

Reply:
Pattern.quote
```java
matcher.appendReplacement(additionalWhereBuilder, Pattern.quote(placeholder));
```

 results in the following string:
```
sale_.name =Q E
```

Reply:
Ah, found the correct one. It's Matcher.quoteReplacement